src/chess/bitboard.h

```diff
@@ -240,10 +264,15 @@ class Move {
   }
 
  private:
-  BoardSquare from_;
-  BoardSquare to_;
-  Promotion promotion_ = Promotion::None;
-  bool castling_ = false;
+  uint16_t data_ = 0;  // First bit castling, next 3 promotion, next 6
```

Comment:
(regarding the comment)
bits are counted from lowest to highest, from zero.
```cpp
// Move, using the following encoding:
// bits 0..5 "to"-square
// bits 6..11 "from"-square
// bits 12..14 promotion value
// bit 15 whether move is a castling
```

Reply:
Done